src/main/java/org/broadinstitute/hellbender/utils/read/MutableGATKRead.java

```diff
@@ -0,0 +1,76 @@
+package org.broadinstitute.hellbender.utils.read;
+
+import htsjdk.samtools.Cigar;
+import htsjdk.samtools.util.Locatable;
+
+/**
+ * Unified read interface for use throughout the GATK.
+ *
+ * Includes operations that allow reads to be modified in-place. Use {@link @GATKRead} if you want immutability.
+ */
+public interface MutableGATKRead extends GATKRead {
+    void setName( final String name );
+
+    void setPosition( final String contig, final int start );
+
+    void setPosition( final Locatable locatable );
+
+    void setMatePosition( final String contig, final int start );
+
+    void setMatePosition( final Locatable locatable );
+
+    void setFragmentLength( final int fragmentLength );
+
+    void setMappingQuality( final int mappingQuality );
+
+    void setBases( final byte[] bases );
+
+    void setBaseQualities( final byte[] baseQualities );
+
+    void setCigar( final Cigar cigar );
+
+    void setCigar( final String cigarString );
+
+    void setReadGroup( final String readGroupID );
+
+    void setNumberOfReadsInFragment( final int numberOfReads );
+
+    void setReadNumber( final int readNumber);
+
+    void setIsPaired( final boolean isPaired );
```

Comment:
I'm not a fan of this naming convention for setters of boolean fields. Suggest `setIsPaired` -> `setPaired`